AdvancementAPI.java

```diff
@@ -44,25 +56,37 @@ public String getIcon() {
         return icon;
     }
 
-    public AdvancementAPI withIcon(String icon) {
+    public MessengerAdvancement withIcon(String icon) {
         this.icon = icon;
         return this;
     }
 
+    public void setIcon(String icon) {
+        this.icon = icon;
+    }
+
     public String getDescription() {
         return description;
     }
 
-    public AdvancementAPI withDescription(String description) {
+    public MessengerAdvancement withDescription(String description) {
         this.description = description;
         return this;
     }
 
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
     public String getBackground() {
         return background;
     }
 
-    public AdvancementAPI withBackground(String url) {
+    public void setBackground(String background) {
```

Comment:
Why renaming it? The name `withBackground` has a reason. It implies thats make use of the BuilderPattern.